teamup-server/src/main/java/com/project/teamup/service/UserService.java

```diff
@@ -32,4 +35,44 @@ public void delete(Long id) {
     public Optional<User> findById(Long id) {
         return userRepository.findById(id);
     }
+
+    /**
+     * Activates a blocked user by setting him
+     * the number of login attempts to 0 and a new generated
+     * password which will be sent by email.
+     * @param admin the admin who activates the blocked user
+     * @param blockedUser the user that becomes a new password
+     *                    via email from the admin
+     * */
+    public String activateUser(User admin, User blockedUser){
+        Optional<User> deactivatedUser = userRepository.findByUsername(blockedUser.getUsername());
+        if(deactivatedUser.isPresent()){
+            User userToBeActivated = deactivatedUser.get();
+            try{
+                String newPassword = generateNewPassword();
+                String subject = MailService.newPasswordSubject();
```

Comment:
It would be better if you would create an enum with all the requirements parts for sending an email, create an generic method in email service and call it just by the enum type + the parameters for the email template